crates/compiler/parse/src/ast.rs

```diff
@@ -933,7 +947,23 @@ impl<'a> Pattern<'a> {
                     false
                 }
             }
-            ListRest => matches!(other, ListRest),
+
+            ListRest(pattern_as) => match other {
+                ListRest(other_pattern_as) => match (pattern_as, other_pattern_as) {
+                    (Some((_, a)), Some((_, b))) => a.equivalent(b),
+                    _ => false,
+                },
+                _ => false,
+            },
+
+            As(pattern, pattern_as) => match other {
+                As(other_pattern, other_pattern_as) => {
+                    pattern_as.equivalent(other_pattern_as)
+                        && pattern.value.equivalent(&other_pattern.value)
+                }
+                _ => false,
+            },
```

Comment:
I'm not 100% sure what kind of equivalence is meant here, but it seems to be stronger than up to renaming? but definitely ignore spaces.